internal/app/server/server.go

```diff
@@ -57,5 +63,17 @@ func Run(ctx context.Context, network string, cfg *config.ServiceConfig) error {
 	pb.RegisterOpenSavesServer(s, server)
 	reflection.Register(s)
 
+	go func() {
+		select {
+		case s := <-sigs:
+			log.Infof("received signal: %v\n", s)
+		case <-ctx.Done():
+			log.Infoln("context cancelled")
+		}
+		healthcheck.SetServingStatus(serviceName, healthgrpc.HealthCheckResponse_NOT_SERVING)
+		s.GracefulStop()
+		log.Infoln("stopping open saves server gracefully")
```

Comment:
This log output should be between L72-73.
Log ("stopping") -> change status -> gracefulstop -> Log ("Stopped")